fix(helm): add support for local version only

technosophos opened the pull request:
This modifies 'helm version' to allow for local-only or server-only
versions to avoid cases where calling 'helm version' was resulting in
errors.

Closes #1440

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/helm/1450)

<!-- Reviewable:end -->

technosophos committed:
fix(helm): add support for local version only

This modifies 'helm version' to allow for local-only or server-only
versions to avoid cases where calling 'helm version' was resulting in
errors.

Closes #1440

adamreese commented:
could the flags be `--client` and `--server`?  if both are set display both rather than return an error message

technosophos committed:
fix(helm): use --client and --server as flags

This changes from --client-only and --server-only to --client and
--server. With this change, if no flags are specified, all info is
shown. If one flag is specified, only its info is shown.

technosophos commented:
@adamreese Done.
